### PR TITLE
Configure security context

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/roadmap/issues/3521

Configure security context for Alloy to run on GiantSwarm clusters.

### CODE REVIEW

File: helm/alloy/values.yaml [hunk 1]
```diff
@@ -10,15 +10,47 @@ networkPolicy:
...
+      runAsGroup: 473
+      runAsNonRoot: true
+      readOnlyRootFilesystem: true
+      seccompProfile:
+        type: RuntimeDefault
 
   serviceMonitor:
     enabled: true
+
+serviceAccount:
```
Comment (QuentinBisson) [inline helm/alloy/values.yaml:54]:
```suggestion
  serviceAccount:
```

Comment (QuentinBisson) [reply helm/alloy/values.yaml:54]: I think that's why the check fails

**Review by QuentinBisson — APPROVED** [review]
LGTM apart from thé service account indentation

File: helm/alloy/values.yaml (hunk 1 above)
Comment (TheoBrigitte) [reply helm/alloy/values.yaml:54]: indentation is wrong, but this does not prevent check from failing

Comment (QuentinBisson) [reply helm/alloy/values.yaml:54]: sure

File: helm/alloy/templates/privileged-rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{{- if and .Values.psp.enabled }}
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
```
Comment (TheoBrigitte) [inline helm/alloy/templates/privileged-rbac.yaml:3]: Wrong we want pss

File: helm/alloy/values.yaml (hunk 1 above)
Comment (TheoBrigitte) [reply helm/alloy/values.yaml]: removed as its no longer needed, and set by default in the upstream chart

**Comment by TheoBrigitte** [comment]
I replaced the psp with kyverno policy exception

**Review by QuentinBisson — APPROVED** [review]

**Pull request merged**